src/ucp/am/eager_multi.c

```diff
@@ -115,7 +115,7 @@ static size_t ucp_am_eager_multi_bcopy_pack_args_mid(void *dest, void *arg)
 
 static UCS_F_ALWAYS_INLINE ucs_status_t ucp_am_eager_multi_bcopy_send_func(
         ucp_request_t *req, const ucp_proto_multi_lane_priv_t *lpriv,
-        ucp_datatype_iter_t *next_iter)
+        ucp_datatype_iter_t *next_iter, ucp_lane_index_t *lane_shift)
```

Comment:
maybe create another function like `ucp_proto_multi_progress` (`ucp_proto_multi_progress_custom_lane` ?), that will not advance `req->send.multi_lane_idx` on its own but let the protocol send callback do it?
- they can probably call each other to avoid code duplication
- extract the code block https://github.com/openucx/ucx/blob/23664bfc3865dcfaab93aaff8dca964ad24c1030/src/ucp/proto/proto_multi.inl#L159-L163 to a new helper function and rndv/get/zcopy will call it
This way will not need to add new parameter and change so many places.